File: collaborationbackend/src/main/java/com/hema/collaborationbackend/service/FriendService.java
package com.hema.collaborationbackend.service;

import java.util.List;

import com.hema.collaborationbackend.model.Friend;
import com.hema.collaborationbackend.model.User;

public interface FriendService {

	List<User> listOfSuggestedUsers(String username);

	void friendRequest(Friend friend);
	
	List<Friend> pendingRequests(String toId);

	void updateFriendRequest(Friend friend);
	
}

File: collaboration_frontend/WebContent/services/friendservice.js
/**
 * 
 */
app.factory('FriendService',function($http){
	
	var friendService={}
	var BASE_URL="http://localhost:8089/CollaborationMiddleware"
		
	friendService.listOfSuggestedUsers=function(){
		return $http.get(BASE_URL + "/getsuggestedusers")
	}
	friendService.sendFriendRequest=function(toId){
		return $http.get(BASE_URL + "/friendrequest/"+toId)
	}
	friendService.pendingRequests=function(){
		return $http.get(BASE_URL + "/pendingrequests")
	}
	
	return friendService;
})